### PR TITLE
Add use index

### LINKED ISSUE #6495: Add _explain endpoint to pouchdb-find
Pouchdb-find is missing the `_explain` endpoint. This will give a details on how the query would be executed.

### PR DESCRIPTION
**Pull request by garrensmith**
This fixes #6495 and #6452

### CODE REVIEW

File: docs/_includes/api/explain_index.html [hunk 1]
```diff
@@ -0,0 +1,135 @@
+{% include anchor.html edit="true" title="Explain index" hash="explain_index" %}
+
+{% highlight js %}
+db.explain(request [, callback])
+{% endhighlight %}
+
+Explai the query plan for a given query
```
Comment (afinne) [inline docs/_includes/api/explain_index.html]: Small typo

Comment (garrensmith) [inline docs/_includes/api/explain_index.html]: Thanks fixed

File: docs/_guides/mango-queries.md [hunk 2]
```diff
@@ -284,14 +284,31 @@ The Mango query language is generally very permissive, and allows you to write q
 
 As a straightforward example, if you query using the `_id` field, then the query planner will automatically map that directly to an `allDocs()` query. However, if you query for a field that isn't yet indexed, then it will simply use `allDocs()` to read in all documents from the database (!) and then filter in-memory. This can lead to poor performance, especially if your database is large.
 
-If you're ever wondering how the query planner is interpreting your query, you can enable debugging like so:
+If you're ever wondering how the query planner is interpreting your query, you can use the explain endpoint:
+
+```js
+db.find({
```
Comment (afinne) [inline docs/_guides/mango-queries.md]: Should this say `.find` ? How does this differ from the normal? Or should this say `.explain` ?

File: docs/_guides/mango-queries.md [hunk 3]
```diff
@@ -301,6 +318,28 @@ You may also want to pay attention to the `"warning"` value included in your res
...
 
+{% include anchor.html title="Set which index to use" hash="use_index" %}
+
+When creating a query, by settings the `use_index` field, it is possible to tell pouchdb-find which index to use.
+The below example shows how to do that.
+
+```js
+db.createIndex({
+  index: {
+    fields: ['age', 'name'],
```
Comment (afinne) [inline docs/_guides/mango-queries.md]: Should this be `['name', 'age']` ?

Comment (garrensmith) [inline docs/_guides/mango-queries.md]: It doesn't matter

File: docs/_guides/mango-queries.md (hunk 2 above)
Comment (garrensmith) [inline docs/_guides/mango-queries.md]: Ah, good catch.

**Pull request merged**